### PR TITLE
[FLINK-35044][state] Init flink-statebackend-forst module

### PR DESCRIPTION
**Pull request by masteryhx**
## What is the purpose of the change

A simple work to introduce a new module for ForStStateBackend, then several subsequent PR e.g. ForStFileSystem, Implementation of Async API, ForStStateBackend could be commited/reviewed in parallel based on it.
See FLIP-427 for more details.

## Brief change log

- Init flink-statebackend-forst module with necessary dependencies

## Verifying this change

This change is a trivial work to introduce a new module without tests coverage. 

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (**yes** / no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (**yes** / no)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by masteryhx** [comment]
@flinkbot run azure

File: flink-state-backends/flink-statebackend-forst/pom.xml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+	<artifactId>flink-statebackend-forst</artifactId>
+	<name>Flink : State backends : ForSt</name>
+
+	<packaging>jar</packaging>
+
+	<properties>
+		<surefire.module.config><!--
+			chill ArraysAsListSerializer
+			-->--add-opens=java.base/java.util=ALL-UNNAMED
+		</surefire.module.config>
```
Comment (ljz2051) [inline flink-state-backends/flink-statebackend-forst/pom.xml:42]: Could you please briefly explain why we need  the "<surefire.module.config>" property?

**Review by ljz2051 — COMMENTED** [review]
Thanks for this PR!  I have one minor question in the comment.

File: flink-state-backends/flink-statebackend-forst/pom.xml (hunk 1 above)
Comment (masteryhx) [reply flink-state-backends/flink-statebackend-forst/pom.xml:42]: It's introduce by FLINK-25002 to supports JDK-17 temporarily.
IIUC, it's better to also introduce it if current dependencies may contains some used modules which should be open, e.g. chill-java in flink-streaming-java.

**Comment by masteryhx** [comment]
@flinkbot run azure

File: flink-state-backends/flink-statebackend-forst/pom.xml (hunk 1 above)
Comment (ljz2051) [reply flink-state-backends/flink-statebackend-forst/pom.xml:42]: Thanks for the explanation! LGTM.

**Review by ljz2051 — APPROVED** [review]

**Pull request closed without merging**